### PR TITLE
feat: #WB-1983: add new component Attachment

### PR DESCRIPTION
**Pull request by Headan**
# Description

Ajout d'un nouveau composant `Attachment` + changement du nom de l'icone Attachment en Paperline

Ticket : #WB-1983
https://edifice-community.atlassian.net/browse/WB-1983

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Which Package changed?

Please check the name of the package you changed

- [x] Components
- [ ] Core
- [x] Icons
- [ ] Hooks

## Has the documentation changed?

- [x] Storybook

## Type of change

Please check options that are relevant.

- [x] Chore (PATCH)
- [x] Doc (PATCH)
- [ ] Bug fix (PATCH)
- [ ] New feature (MINOR)
- [ ] Breaking change (MAJOR)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: packages/react/src/components/Attachment/Attachment.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
+import { Meta, StoryObj } from "@storybook/react";
+
+import { Button, IconButton } from "../Button";
+import Attachment, { AttachmentProps } from "./Attachment";
+import { Delete, Download } from "@edifice-ui/icons";
+import { Grid } from "../Grid";
+
+export default {
+  title: "Components/AttachmentComponent",
```
Comment (ccreusat) [inline packages/react/src/components/Attachment/Attachment.stories.tsx:9]: @Headan 

- Renommer la story "Attachment"

File: packages/react/src/components/Attachment/Attachment.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  component: Attachment,
+  subcomponents: { Button },
```
Comment (ccreusat) [inline packages/react/src/components/Attachment/Attachment.stories.tsx:11]: @Headan 
Retirer le champ `subcomponent` car ne sert plus à rien avec SB7

File: packages/react/src/components/Attachment/Attachment.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+export type AttachmentType = React.FC<AttachmentProps>;
+
+const Attachment = forwardRef(
+  (
+    { name = "Attachment Name", options, ...restProps }: AttachmentProps,
+    ref: Ref<HTMLDivElement>,
+  ) => {
+    return (
+      <div ref={ref} className="attachment px-8 py-12" {...restProps}>
```
Comment (ccreusat) [inline packages/react/src/components/Attachment/Attachment.tsx:24]: @Headan il faut inverser les valeurs de padding, c'est : 

- px-12
- py-8

vu avec simon :)

File: packages/react/src/components/Attachment/Attachment.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        <div className="filename">
+          <Paperclip height={22} width={22} />
+          <p className="name">{name}</p>
```
Comment (ccreusat) [inline packages/react/src/components/Attachment/Attachment.tsx:27]: - remplacer le `className="name"` par `className="body"` 

ça prendra bien la font 16px et le line-height 24px

File: packages/react/src/components/Attachment/Attachment.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        </div>
+        <div className="options px-2">{options}</div>
```
Comment (ccreusat) [inline packages/react/src/components/Attachment/Attachment.tsx:29]: vu avec simon, c'est `px-12`

**Review by ccreusat — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Attachment/Attachment.tsx (hunk 4 above)
Comment (Headan) [reply packages/react/src/components/Attachment/Attachment.tsx:27]: @ccreusat Le problème c'est que j'ai rajouter des propriétés css pour name :/ mais je peux changer son name

Comment (ccreusat) [reply packages/react/src/components/Attachment/Attachment.tsx:27]: `body text-truncate`

**Review by ccreusat — APPROVED** [review]

**Pull request merged**